File: backend/lib/bankfacil/order_item.rb

module BankFacil
	class OrderItem
		attr_reader :order, :product
		attr_accessor :quantity

		def initialize(order, product, quantity)
			@order = order
			@product = product
			@quantity = quantity
		end

		def total
			@product.price * @quantity
		end

		def ship?
			@product.ship?
		end

		def tax_free?
			@product.tax_free?
		end

		def when_order_closed
			@product.when_order_closed(order)
		end
	end
end

File: backend/BankFacil/OrderItem.cs
namespace BankFacil
{
    public class OrderItem
    {
        public Order Order { get; }
        public Product Product { get; }

        public int Total { get { return 10; } }

        public OrderItem(Order order, Product product)
        {
            Order = order;
            Product = product;
        }
    }
}
